content/en/docs/concepts/scheduling-eviction/api-eviction.md

```diff
@@ -6,14 +6,117 @@ weight: 70
+POST the attempted operation, similar to the following example:
+
+{{< tabs name="Eviction_example" >}}
+{{% tab name="policy/v1" %}}
+{{< note >}}
+`policy/v1` Eviction is available in v1.22+. Use `policy/v1beta1` with prior releases.
+{{< /note >}}
+
+```json
+{
+  "apiVersion": "policy/v1",
+  "kind": "Eviction",
+  "metadata": {
+    "name": "quux",
+    "namespace": "default"
+  }
+}
+```
+{{% /tab %}}
+{{% tab name="policy/v1beta1" %}}
+{{< note >}}
+Deprecated in v1.22 in favor of `policy/v1`
+{{< /note >}}
+
+```json
+{
+  "apiVersion": "policy/v1beta1",
+  "kind": "Eviction",
+  "metadata": {
+    "name": "quux",
+    "namespace": "default"
+  }
+}
+```
+{{% /tab %}}
+{{< /tabs >}}
+
+Alternatively, you can attempt an eviction operation by accessing the API using
+`curl` or `wget`, similar to the following example:
+
+```bash
+curl -v -H 'Content-type: application/json' https://your-cluster-api-endpoint.example/api/v1/namespaces/default/pods/quux/eviction -d @eviction.json
+```
+
+## How API-initiated eviction works
+
+When you request an eviction using the API, the API server performs admission
+checks and responds in one of the following ways:
+
+* `200 OK`: the eviction is allowed, the `Eviction` subresource is created, and
+  the Pod is deleted, similar to sending a `DELETE` request to the Pod URL.
+* `429 Too Many Requests`: the eviction is not currently allowed because of the
+  configured {{<glossary_tooltip term_id="pod-disruption-budget" text="PodDisruptionBudget">}}.
+  You may be able to attempt the eviction again later. You might also see this
+  response because of API rate limiting. 
+* `500 Internal Server Error`: the eviction is not allowed because there is a
+  misconfiguration, like if multiple PodDisruptionBudgets reference the same Pod.
+
+If the Pod you want to evict isn't part of a workload that has a
+PodDisruptionBudget, the API server always returns `200 OK` and allows the
+eviction. 
+
+If the API server allows the eviction, the Pod is deleted as follows:
+
+1. The `Pod` resource in the API server is updated with a deletion timestamp,
+   after which the API server considers the `Pod` resource to be terminated. The
+   `Pod` resource is also marked with the configured grace period.
+1. The {{<glossary_tooltip term_id="kubelet" text="kubelet">}} on the node where the local Pod is running notices that the `Pod`
+   resource is marked for termination and starts to gracefully shut down the
+   local Pod.
+1. While the kubelet is shutting the Pod down, the control plane removes the Pod
+   from {{<glossary_tooltip term_id="endpoint" text="Endpoint">}} and
+   {{<glossary_tooltip term_id="endpoint-slice" text="EndpointSlice">}}
+   objects. As a result, controllers no longer consider the Pod as a valid object.
+1. After the grace period for the Pod expires, the kubelet forcefully terminates
+   the local Pod.
+1. The kubelet tells the API server to remove the `Pod` resource.
```

Comment:
Are API evictions special cased? - In the case of a kubelet-initiated eviction the Pod gets an `Evicted` status, but still then needs to eventually be explicitly deleted separately from the eviction process

Reply:
I'm not sure whether the kubelet or some control plane component sends the **DELETE** request / performs the equivalent operation.

It's possible that API-initiated eviction lists the kubelet as a finalizer, and the action that the kubelet takes is to remove its finalizer once the Pod has terminated (gracefully or otherwise). In that case, it's the API server that spots there is a resource pending deletion with no finalizer remaining, and triggers cleanup.

SIG Node are likely to be the best folks to confirm this, as they look after both the kubelet code and (most of) the Pod API.

Reply:
I'm part of sig-node 😅  (and have spent a lot of time looking at kubelet-initiated eviction code).

I've been meaning to actually test this (or look into api-server code) to see what happens, but haven't got around to it yet bc of other priorities. I'll try and look next week,